src/components/Comment.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { formatDistance, subDays } from 'date-fns';
+import CommentForm from './CommentForm.tsx';
+
+type CommentTypes = {
+  id: string;
+  username: string;
+  image: string;
+  body: string;
+  userId: string;
+  createdAt: string;
+};
+type ActiveComment = {
+  id: string;
+  type: string; // You can specify more specific types for 'type' if needed
+};
+type CommentProps = {
+  comment: CommentTypes;
+  replies: Array<CommentTypes>;
+  setActiveComment: (comment: ActiveComment | null) => void;
```

Comment:
instead of passing the state setter, we could make this component more dumb by replacing this by some event emitters

```suggestion
  onEditCommentClick: (commentId: ActiveComment['id']) => void;
  onEditCommentSubmit: (comment: ActiveComment) => void;
  onReplay: (replay: ActiveComment, commentId: ActiveComment['id']) => void;
```